api/v3/clustermanager_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2018-2022 Splunk Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v3
+
+import (
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// default all fields to being optional
+// +kubebuilder:validation:Optional
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+// see also https://book.kubebuilder.io/reference/markers/crd.html
+
+const (
+	// ClusterManagerPausedAnnotation is the annotation that pauses the reconciliation (triggers
+	// an immediate requeue)
+	ClusterManagerPausedAnnotation = "clustermanager.enterprise.splunk.com/paused"
+)
+
+// ClusterManagerSpec defines the desired state of ClusterManager
+type ClusterManagerSpec struct {
+	CommonSplunkSpec `json:",inline"`
+
+	// Splunk Smartstore configuration. Refer to indexes.conf.spec and server.conf.spec on docs.splunk.com
+	SmartStore SmartStoreSpec `json:"smartstore,omitempty"`
+
+	// Splunk Enterprise App repository. Specifies remote App location and scope for Splunk App management
+	AppFrameworkConfig AppFrameworkSpec `json:"appRepo,omitempty"`
+}
+
+// ClusterManagerStatus defines the observed state of ClusterManager
+type ClusterManagerStatus struct {
+	// current phase of the cluster manager
+	Phase splcommon.Phase `json:"phase"`
+
+	// selector for pods, used by HorizontalPodAutoscaler
+	Selector string `json:"selector"`
+
+	// Splunk Smartstore configuration. Refer to indexes.conf.spec and server.conf.spec on docs.splunk.com
+	SmartStore SmartStoreSpec `json:"smartstore,omitempty"`
+
+	// Bundle push status tracker
+	BundlePushTracker BundlePushInfo `json:"bundlePushInfo"`
+
+	// Resource Revision tracker
+	ResourceRevMap map[string]string `json:"resourceRevMap"`
+
+	// App Framework status
+	AppContext AppDeploymentContext `json:"appContext"`
+}
+
+// BundlePushInfo Indicates if bundle push required
```

Comment:
==> Key point for review, this is a single exception:

We can't get rid of `needToPushMasterApps` because existing deployments already use it, and we can't allow new deployments to use it moving forward. 

My solution was to enable`BundlePushInfo` to be shared between Master and Manager.  
```
Master code uses `needToPushMasterApps`
Manager code uses `needToPushManagerApps`
```
Once the Master CRD is removed we can also remove this exception.

CC: @sgontla , @vivekr-splunk , @gaurav-splunk